src/Kmd.Logic.Consumption.Client.AuditClient/AuditClientConsumptionMetricsDestination.cs

```diff
@@ -34,30 +34,32 @@ public IConsumptionMetricsDestination ForSubscriptionOwnerContext(string propert
 
         public void Write(Guid subscriptionId, Guid resourceId, string meter, int amount, string reason = null)
         {
-            var audit = reason == null
-                        ? this._audit
-                        : this._audit.ForContext("Reason", reason);
-
-            audit.Write(
-                Template,
-                amount,
-                meter,
-                resourceId,
-                subscriptionId);
+            this.Write(
+                subscriptionId: subscriptionId,
+                resourceId: resourceId,
+                meter: meter,
+                amount: amount,
+                ConsumedDatetime: DateTimeOffset.Now,
+                reason: reason);
         }
 
-        public void Write(string propertyName, Guid subscriptionId, Guid resourceId, string meter, int amount, DateTimeOffset logConsumedDatetime, string reason = null)
+        public void Write(Guid subscriptionId, Guid resourceId, string meter, int amount, DateTimeOffset ConsumedDatetime, string reason = null)
         {
             var audit = reason == null
                        ? this._audit
-                       : this._audit.ForContext(this._getSubOwnerContextName(propertyName), reason);
+                       : this._audit.ForContext(this._getSubOwnerContextName("Reason"), reason);
+
+            audit = ConsumedDatetime == null
+                      ? audit
+                      : audit.ForContext(this._getSubOwnerContextName("LogConsumedDatetime"), ConsumedDatetime);
+
             audit.Write(
                 Template,
```

Comment:
Will need to use a different template here. The new template will include the date property name, so that the additional date argument is correctly matched to a property name, and does appear as a numerical named value in the resulting event.